src/Stack/Ghci.hs

```diff
@@ -957,17 +960,19 @@ makeGhciPkgInfo installMap installedMap locals addPkgs mfileTargets pkgDesc = do
 wantedPackageComponents :: BuildOpts -> Target -> Package -> Set NamedComponent
 wantedPackageComponents _ (TargetComps cs) _ = cs
 wantedPackageComponents bopts (TargetAll PTProject) pkg = S.fromList $
-     ( case packageLibraries pkg of
-         NoLibraries -> []
-         HasLibraries names -> CLib : map CSubLib (S.toList names)
-     )
-  <> map CExe (S.toList (packageExes pkg))
-  <> map CSubLib (S.toList $ packageSubLibraries pkg)
-  <> (if boptsTests bopts then map CTest (M.keys (packageTests pkg)) else [])
-  <> ( if boptsBenchmarks bopts
-         then map CBench (S.toList (packageBenchmarks pkg))
+     ( if hasMainBuildableLibrary pkg
+         then CLib : map CSubLib buildableForeignLibs
          else []
      )
+  <> map CExe (S.toList (packageExes pkg))
+  <> map CSubLib buildableSubLibs
+  <> (if boptsTests bopts then map CTest buildableTestSuites else [])
+  <> (if boptsBenchmarks bopts then map CBench buildableBenchmarks else [])
+ where
+  buildableForeignLibs = getBuildableListText $ packageForeignLibraries pkg
+  buildableSubLibs = getBuildableListText $ packageSubLibraries pkg
+  buildableTestSuites = getBuildableListText $ packageTestSuites pkg
+  buildableBenchmarks = getBuildableListText $ packageBenchmarkSuites pkg
```

Comment:
When I refactored to create these four lines (to avoid long lines above), I realised that we now have `Stack.Package.packageSubLibrariesNameSet` (which I would rename `buildableSubLibs`) and `Stack.Package.packageBenchmarks` (which I would rename `buildableBenchmarks`). I was wondering: should we be consistent and have five functions `Stack.Package.buildable<component_type> :: Package -> Set Text` for each of the component types? (So `packageExes` gets renamed `buildableExes`.)

Reply:
yes thaat sounds good !